packages/core/src/template-compiler-common.ts

```diff
@@ -224,7 +224,20 @@ export class TemplateCompiler {
 }
 
 function matchesSourceFile(filename: string) {
```

Comment:
does this function have a test?

Reply:
Added a basic test to cover this. We can restructure this in the future to be more as you describe.